spring-integration-core/src/main/java/org/springframework/integration/endpoint/PollingConsumer.java

```diff
@@ -71,15 +91,88 @@ protected void doStop() {
 
 	@Override
 	protected void handleMessage(Message<?> message) {
-		this.handler.handleMessage(message);
+		boolean executorInterceptorPresent = false;
+		if (this.channelInterceptors != null) {
+			for (ChannelInterceptor interceptor : this.channelInterceptors) {
+				if (interceptor instanceof ExecutorChannelInterceptor) {
+					executorInterceptorPresent = true;
```

Comment:
Sounds reasonable.
Will try and come back to you tomorrow (or in the evening) on the matter.
Thanks